src/connectors/pagination/__tests__/connectPagination-test.js

```diff
@@ -361,113 +380,76 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/pagination/
     });
   });
 
-  describe('routing', () => {
-    const getInitializedWidget = () => {
-      const renderFn = jest.fn();
-      const makeWidget = connectPagination(renderFn);
-      const widget = makeWidget({});
-
-      const helper = algoliasearchHelper({}, '');
-      helper.search = jest.fn();
-
-      widget.init({
+  describe('getWidgetState', () => {
+    test('should give back the object unmodified if the default value is selected', () => {
+      const [widget, helper] = getInitializedWidget();
+      const uiStateBefore = {};
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
         helper,
-        state: helper.state,
-        createURL: () => '#',
       });
 
-      const { refine } = renderFn.mock.calls[0][0];
+      expect(uiStateAfter).toBe(uiStateBefore);
```

Comment:
We don't care about it being the same reference. We only want to check the value, right?

Reply:
At some point IIRC it was important. But anyway my PR move this code but didn't change it at all. Only the code related to `getWidgetSearchParameters` changes (implem + tests).